discovery_engine_core/core/src/document.rs

```diff
@@ -129,7 +129,7 @@ pub struct UserReacted {
     pub snippet: String,
 
     /// Precomputed S-mBert of the document.
-    pub smbert: Embedding,
+    pub smbert_embedding: Embedding,
```

Comment:
while we are at it we could also rename the `id` field in l.123 to `document_id` to better differentiate it from the `stack_id` field and to align the namings in the ffi.

Reply:
The type is already `Embedding` why we need to repeat it?

Reply:
field's named `id` are the id of the entity they are in. I don't think will be necessary to additionally add `document_` to avoid stuff like `document.document_id`, `document.id` seems already clear.

Reply:
> field's named `id` are the id of the entity they are in. I don't think will be necessary to additionally add `document_` to avoid stuff like `document.document_id`, `document.id` seems already clear.

true, i haven't looked at it this way.

Reply:
I called it `smbert_embedding` because it is named that way in other places. And it's not a `smbert`. Sure field names don't have to repeat the type. But IMHO the parts you can cut out are the non-stem parts. For S-mBert Embeddings the stem is Embedding not S-mBert. (And yes that linguistically seen probably not a fully correct usage of the word stem, I hope it still conveys what I mean.)